pennylane/templates/subroutines/qsvt.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the QSVT template and qsvt wrapper function.
+"""
+import numpy as np
+import pennylane as qml
+from pennylane.queuing import QueuingManager
+from pennylane.ops import BlockEncode, PCPhase
+from pennylane.ops.op_math import adjoint
+from pennylane.operation import AnyWires, Operation
+
+
+def qsvt(A, phi_vect, wires):
+    """Performs the
+    `quantum singular value transformation <https://arxiv.org/abs/1806.01838>`__ circuit using
+    :class:`~.BlockEncode` and :class:`~.PCPhase`.
+
+    Args:
+        A (Array): the general :math:`(n \times m)` matrix to be encoded.
+        phi_vect (list[Float]): a list of angles by which to shift to obtain the desired polynomial.
+        wires (Iterable): the wires the template acts on.
```

Comment:
- Perhaps `A` can be `tensor_like` if it's possible to pass in, e.g., a Torch tensor
- `list[Float]` -> `Sequence[float]` (unless you're explicitly requiring it's a list). Or should it be `tensor_like` if people can differentiate with respect to it?

Reply:
Changed both to tensor_like.

We can differentiate with respect to both and both should be able to accept e.g. a Torch tensor.